From 7f831482b8e55b772aa23e93a84dc1e2721ec719 Mon Sep 17 00:00:00 2001
From: Gerrit 'Geaz' Gazic <geaz@users.noreply.github.com>
Date: Mon, 27 Apr 2020 23:19:18 +0200
Subject: [PATCH] Windows: Hide window until run

Because the window is shown as soon as the webview gets created, it is possible, that the end users see a 'flickering' window. For example, if the developer sets the size of the window, first a window is shown with default sizes and no content, then it gets resized (flickering).

This commit creates a 'hidden' window on initialization and it gets shown on calling the run method. In between of the initialization and running the webview the change of configuration parameters, through calling methods like set_title and set_size, will only affect the hidden window and it will not occur as 'flickering' to the user.
---
 webview.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/webview.h b/webview.h
index 20ebe07..9959938 100644
--- a/webview.h
+++ b/webview.h
@@ -982,7 +982,7 @@ class win32_edge_engine {
       m_window = *(static_cast<HWND *>(window));
     }
 
-    ShowWindow(m_window, SW_SHOW);
+    ShowWindow(m_window, SW_HIDE);
     UpdateWindow(m_window);
     SetFocus(m_window);
 
@@ -998,6 +998,8 @@ class win32_edge_engine {
   }
 
   void run() {
+    ShowWindow(m_window, SW_SHOW);
+    
     MSG msg;
     BOOL res;
     while ((res = GetMessage(&msg, nullptr, 0, 0)) != -1) {
